orchestra/static/orchestra/todos/todo-list.directive.es6.js

```diff
@@ -98,9 +99,26 @@ export default function todoList (orchestraApi) {
         todoApi.delete(todo)
       }
 
+      todoList.addActionToTodoActivityLog = (todo, action) => {
+        const datetimeUtc = moment.tz(moment(), moment.tz.guess()).utc()
+        var activityLog = JSON.parse(todo.activity_log.replace(/'/g, '"'))
+        activityLog['actions'].push({
+          'action': action,
+          'datetime': datetimeUtc.format('YYYY-MM-DD HH:mm:ss')
+        })
+        todo.activity_log = JSON.stringify(activityLog).replace(/'/g, '"')
+      }
+
+      todoList.onToggleTodo = (todo, collapsed) => {
+        todoList.addActionToTodoActivityLog(todo, collapsed ? 'collapse' : 'expand')
+        todoApi.update(todo)
+      }
+
       todoList.skipTodo = (todo) => {
         const datetimeUtc = moment.tz(moment(), moment.tz.guess()).utc()
         todo.skipped_datetime = datetimeUtc.format('YYYY-MM-DD HH:mm')
+        // Not passing todo.skipped_datetime because we want to log datetime at a higher resolution (includes seconds).
```

Comment:
that's funny :). I'd have argued we should have high-resolution data in both :)

Reply:
Sanity check
- [x] Old skipped todos works fine.
- [x] Due date functionality works fine.